pages/3_monster_detail.py

```diff
@@ -61,7 +61,7 @@ def reformat_breeding_list(json_data, monster_page_id):
 
     params = st.experimental_get_query_params()
     if "id" in params:
-        monster_id = params["id"][0]
+        monster_id = int(params["id"][0])
```

Comment:
I liked what we found about typing and possibly mypy catching it preemptively here